src/Evolution/Systems/NewtonianEuler/BoundaryConditions/DirichletAnalytic.hpp

```diff
@@ -0,0 +1,137 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <memory>
+#include <optional>
+#include <pup.h>
+#include <string>
+#include <type_traits>
+
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Evolution/BoundaryConditions/Type.hpp"
+#include "Evolution/Systems/NewtonianEuler/BoundaryConditions/BoundaryCondition.hpp"
+#include "Evolution/Systems/NewtonianEuler/ConservativeFromPrimitive.hpp"
+#include "Evolution/Systems/NewtonianEuler/Fluxes.hpp"
+#include "Evolution/Systems/NewtonianEuler/Tags.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "PointwiseFunctions/AnalyticData/Tags.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/AnalyticSolution.hpp"
+#include "Time/Tags.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace domain::Tags {
+template <size_t Dim, typename Frame>
+struct Coordinates;
+}  // namespace domain::Tags
+/// \endcond
+
+namespace NewtonianEuler::BoundaryConditions {
+/*!
+ * \brief Sets Dirichlet boundary conditions using the analytic solution or
+ * analytic data.
+ */
+template <size_t Dim>
+class DirichletAnalytic final : public BoundaryCondition<Dim> {
```

Comment:
(general question, not specific to this PR) When a new person wants to add a boundary condition to an evolution system, where could they go to understand the interface they have to implement? The system-specific base class doesn't (and probably isn't the right place to) document the expected `bc_type` static member variable, the type aliases like `dg_interior_evolved_variables_tags` and the member functions like `dg_ghost`. Perhaps the system-specific base-classes could derive off a common base class that describes the expected interface? I think it would be nice if one could find the documentation starting from a boundary-condition class like this, e.g. by tracing the inheritance tree.

Reply:
there is not. At some point I'll write up a general document detailing how we do DG, including the different interfaces for this pattern, but until the code doing that is actually merged and finalized I don't really feel like putting in the effort to do a big write up.